src/components/MobileWalletDialog/routes/ImportWallet/ImportKeystore.tsx

```diff
@@ -0,0 +1,174 @@
+import {
+  Box,
+  Button,
+  FormControl,
+  FormErrorMessage,
+  Input,
+  Text as CText,
+  VStack,
+} from '@chakra-ui/react'
+import { useCallback, useState } from 'react'
+import type { FieldValues } from 'react-hook-form'
+import { useForm } from 'react-hook-form'
+import { useTranslate } from 'react-polyglot'
+import { useHistory } from 'react-router-dom'
+import { MobileWalletDialogRoutes } from 'components/MobileWalletDialog/types'
+import { DialogBackButton } from 'components/Modal/components/DialogBackButton'
+import { DialogBody } from 'components/Modal/components/DialogBody'
+import { DialogCloseButton } from 'components/Modal/components/DialogCloseButton'
+import { DialogFooter } from 'components/Modal/components/DialogFooter'
+import {
+  DialogHeader,
+  DialogHeaderLeft,
+  DialogHeaderRight,
+} from 'components/Modal/components/DialogHeader'
+import { SlideTransition } from 'components/SlideTransition'
+import { Text } from 'components/Text'
+import { addWallet } from 'context/WalletProvider/MobileWallet/mobileMessageHandlers'
+import { FileUpload } from 'context/WalletProvider/NativeWallet/components/NativeImportKeystore'
+import type { NativeWalletValues } from 'context/WalletProvider/NativeWallet/types'
+import { getMixPanel } from 'lib/mixpanel/mixPanelSingleton'
+import { MixPanelEvent } from 'lib/mixpanel/types'
+
+export const ImportKeystore = () => {
+  const history = useHistory()
+  const [keystoreFile, setKeystoreFile] = useState<string | null>(null)
+  const mixpanel = getMixPanel()
+
+  const translate = useTranslate()
+
+  const {
+    setError,
+    handleSubmit,
+    formState: { errors, isSubmitting, isValid },
+    register,
+  } = useForm<NativeWalletValues>({
+    mode: 'onChange',
+    shouldUnregister: true,
+  })
+
+  const onSubmit = useCallback(
+    async (values: FieldValues) => {
+      const { Vault } = await import('@shapeshiftoss/hdwallet-native-vault')
+      const vault = await Vault.create(undefined, false)
+      vault.meta.set('createdAt', Date.now())
+
+      if (!keystoreFile) {
+        throw new Error('No keystore uploaded')
+      }
+
+      try {
+        await vault.loadFromKeystore(keystoreFile, values.keystorePassword)
+      } catch (e) {
+        setError('keystorePassword', {
+          type: 'manual',
+          message: translate('walletProvider.shapeShift.import.invalidKeystorePassword'),
+        })
+        return
+      }
+
+      try {
+        const mnemonic = await vault.unwrap().get('#mnemonic')
+
+        const revocableVault = await addWallet({
+          mnemonic,
+          label: values.name.trim(),
+        })
+
+        vault.revoke()
```

Comment:
blocking: can we use `decryptFromKeystore` method to achieve this vs. creating/revoking a dummy wallet? i.e
```typescript
import { decryptFromKeystore } from '@shapeshiftoss/hdwallet-native-vault'
```

Reply:
This didn't work, I did port this method to web adapting it as discussed in private

in [b0dae5c](https://github.com/shapeshift/web/pull/8848/commits/b0dae5cef385e8876b2434989d8da73dca1c2b25)